airbyte-integrations/connectors/source-myhours/source_myhours/source.py

```diff
@@ -0,0 +1,172 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+import json
+import logging
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from urllib.parse import parse_qs, urlparse
+
+import pendulum
+import requests
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
+
+from .constants import request_headers, url_base
+
+
+class MyhoursStream(HttpStream, ABC):
+    url_base = url_base + "/"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        json = response.json()
+        for record in json:
+            yield record
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        return request_headers
+
+
+class Clients(MyhoursStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "Clients"
+
+
+class Projects(MyhoursStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "Projects/getAll"
+
+
+class Tags(MyhoursStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "Tags"
+
+
+class TimeLogs(MyhoursStream):
```

Comment:
While I did consider this, I indeed chose to do full refresh for now.

Things holding me back:
- This is my first OSS pull-request, first time using Python and thus like to keep it simple for now. Thankfully the CDK made it simple for me
- The My Hours API does not specify whether a log is still running (using the timer function) or completed. Time logs can still be running through the next day. Even though this log will only show up in the API response of the previous day. 
- Time logs can be edited in myhours. There's no API that can list the last changed time logs and there's no updated_at filed on time logs

Since implementing an incremental sync mode is a bit more advanced and the My Hours API has these quirks I opted for only creating full refresh. If you see obvious solutions to these problems and can give pointers to implement incremental mode without too much data loss then please let me know. If you or someone on your team could outline how to circumvent the above points then I could take a look at getting it to work.

Reply:
All right, thanks for the details, this can be part of a future improvement if the source API makes it possible.